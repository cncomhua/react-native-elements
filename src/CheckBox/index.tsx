import { withTheme } from '../config';
import { CheckBox, CheckBoxProps } from './CheckBox';

export { CheckBox };
export type { CheckBoxProps };
/**
 * RNE CheckBox
 *
 * **Preview**
 *
 * ![URL](URL)
 *
 * [**Documentation**](https://reactnativeelements.com/docs/checkbox)
 */
export default withTheme(CheckBox, 'Checkbox');
