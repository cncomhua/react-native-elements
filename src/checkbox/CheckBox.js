import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, View, Text as NativeText } from 'react-native';
import TextElement from '../text/Text';
import FAIcon from 'react-native-vector-icons/FontAwesome';
import getIconType from '../helpers/getIconType';
import ViewPropTypes from '../config/ViewPropTypes';

class CheckBox extends React.PureComponent {
  render() {
    const {
      component,
      checked,
      iconRight,
      title,
      center,
      right,
      containerStyle,
      textStyle,
      onPress,
      onLongPress,
      onIconPress,
      onLongIconPress,
      checkedIcon,
      uncheckedIcon,
      iconType,
      checkedColor,
      uncheckedColor,
      checkedTitle,
      fontFamily,
      ...attributes
    } = this.props;

    let Icon = FAIcon;
    if (iconType) {
      Icon = getIconType(iconType);
    }
    const Component = component || TouchableOpacity;
    let iconName = uncheckedIcon;
    if (checked) {
      iconName = checkedIcon;
    }

    const styles = this.context.theme.checkbox;

    return (
      <Component
        onLongPress={onLongPress}
        onPress={onPress}
        style={[styles.container, containerStyle && containerStyle]}
        {...attributes}
      >
        <View
          style={[
            styles.wrapper,
            right && { justifyContent: 'flex-end' },
            center && { justifyContent: 'center' },
          ]}
        >
          {!iconRight &&
            <Icon
              color={checked ? checkedColor : uncheckedColor}
              name={iconName}
              size={24}
              onLongPress={onLongIconPress}
              onPress={onIconPress}
            />}
          <TextElement
            style={[
              styles.text,
              textStyle && textStyle,
              fontFamily && { fontFamily },
            ]}
          >
            {checked ? checkedTitle || title : title}
          </TextElement>
          {iconRight &&
            <Icon
              color={checked ? checkedColor : uncheckedColor}
              name={iconName}
              size={24}
              onLongPress={onLongIconPress}
              onPress={onIconPress}
            />}
        </View>
      </Component>
    );
  }
}

CheckBox.defaultProps = {
  checked: false,
  iconRight: false,
  right: false,
  center: false,
  checkedColor: 'green',
  uncheckedColor: '#bfbfbf',
  checkedIcon: 'check-square-o',
  uncheckedIcon: 'square-o',
};

CheckBox.propTypes = {
  component: PropTypes.any,
  checked: PropTypes.bool,
  iconRight: PropTypes.bool,
  title: PropTypes.string,
  center: PropTypes.bool,
  right: PropTypes.bool,
  containerStyle: ViewPropTypes.style,
  textStyle: NativeText.propTypes.style,
  onPress: PropTypes.func,
  onLongPress: PropTypes.func,
  checkedIcon: PropTypes.string,
  uncheckedIcon: PropTypes.string,
  iconType: PropTypes.string,
  checkedColor: PropTypes.string,
  uncheckedColor: PropTypes.string,
  checkedTitle: PropTypes.string,
  onIconPress: PropTypes.func,
  onLongIconPress: PropTypes.func,
  fontFamily: PropTypes.string,
};

CheckBox.contextTypes = {
  theme: PropTypes.object.isRequired,
};

export default CheckBox;
