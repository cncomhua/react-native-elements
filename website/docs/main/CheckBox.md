---


id: checkbox


title: CheckBox


slug: /checkbox

---



import Usage from './usage/CheckBox/CheckBox.md'



CheckBoxes allow users to complete tasks that involve making choices such as selecting options, or switching settings - On or Off.

It provides a clear visual of either a true or false choice.



## Usage


<Usage />

---


## Props

### CheckBox




## Reference

### CheckBox

