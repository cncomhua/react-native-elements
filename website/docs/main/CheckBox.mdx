---
id: checkbox

title: CheckBox

slug: /checkbox
---

import Usage from './usage/CheckBox/CheckBox.mdx';

import Props from './props/CheckBox.mdx';

import { CheckBox } from 'react-native-elements';

import Tabs from '@theme/Tabs';

import TabItem from '@theme/TabItem';

CheckBoxes allow users to complete tasks that involve making choices such as selecting options, or switching settings - On or Off.

It provides a clear visual of either a true or false choice.

## Usage

<Usage />

## Props

<Props />
