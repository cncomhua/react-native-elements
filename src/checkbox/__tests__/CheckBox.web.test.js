import React from 'react';
import {
  Text,
  TouchableHighlight,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../../../samples';
import CheckBox from '../CheckBox';

const props = {
  noProps: {
    component: CheckBox,
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'component, title, checked, & onIconPress': {
    component: CheckBox,
    props: {
      component: TouchableWithoutFeedback,
      checked: true,
      title: 'Title',
      onIconPress: () => {
        let x = 7;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      getJsxString: () => {
        return `const TouchableWithoutFeedback = RN.TouchableWithoutFeedback;
  <CheckBox 
    component={TouchableWithoutFeedback}
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
  />`;
      },
    },
  },
  'iconRight, title, checked, & onIconPress': {
    component: CheckBox,
    props: {
      iconRight: true,
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
    iconRight
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
  />`;
      },
    },
  },
  'right, title, checked, & onIconPress': {
    component: CheckBox,
    props: {
      right: true,
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
  right
  title="title"
  checked={state.checked}
  onIconPress={() => {
    setState({ checked: !state.checked });
  }}
/>`;
      },
    },
  },
  'center, title, checked, & onIconPress': {
    component: CheckBox,
    props: {
      center: true,
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
    center
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
  />`;
      },
    },
  },
  containerStyle: {
    component: CheckBox,
    props: { containerStyle: { backgroundColor: '#071' } },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'textStyle & title': {
    component: CheckBox,
    props: { textStyle: { backgroundColor: '#071' }, title: 'Title' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'onLongPress, title, checked, & onIconPress': {
    component: CheckBox,
    props: {
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
      onLongPress: () => {
        let y = 8;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
    onLongPress={() => {
      console.log("pressed, long");
    }}
  />`;
      },
    },
  },
  'onLongIconPress, title, checked, & onIconPress': {
    component: CheckBox,
    props: {
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
      onLongPress: () => {
        let y = 8;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
    onLongIconPress={() => {
      console.log("pressed, long");
    }}
  />`;
      },
    },
  },
  'onPress, title, checked, & onIconPress': {
    component: CheckBox,
    props: {
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
      onPress: () => {
        let y = 8;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
    onPress={() => {
      setState({ checked: !state.checked });
    }}
  />`;
      },
    },
  },
  checkedIcon: {
    component: CheckBox,
    props: { checked: true, checkedIcon: 'bell-o' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  uncheckedIcon: {
    component: CheckBox,
    props: { uncheckedIcon: 'bell-slash-o' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  checkedColor: {
    component: CheckBox,
    props: { checked: true, checkedColor: '#22d' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  uncheckedColor: {
    component: CheckBox,
    props: { uncheckedColor: '#d22' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'checkedTitle, title, checked, & onIconPress': {
    component: CheckBox,
    props: {
      checkedTitle: 'title - is checked',
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
    checkedTitle="title - is checked"
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
  />`;
      },
    },
  },
  fontFamily: {
    component: CheckBox,
    props: { title: 'Title', fontFamily: 'Courier New' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Input: { sectionComponents: { CheckBox: { samples } } },
  },
  options
);

describe('CheckBox Component', () => {
  it('should render without issues', () => {
    const component = shallow(<CheckBox />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should use TouchableOpacity as default component', () => {
    const component = shallow(<CheckBox />);

    expect(component.find('TouchableOpacity').length).toBe(1);
  });

  it('should allow to pass custom component', () => {
    const View = jest.fn();
    const component = shallow(<CheckBox component={View} />);

    expect(component.find('View').length).toBe(1);
  });

  it('should render title in Text', () => {
    const component = shallow(<CheckBox title="Custom Text" checked />);

    expect(component.props().children.props.children[1].props.children).toBe(
      'Custom Text'
    );
  });

  it('should render with icon and checked', () => {
    const component = shallow(
      <CheckBox
        iconType="font-awesome"
        checkedColor="red"
        containerStyle={{ backgroundColor: 'red' }}
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with icon and iconRight', () => {
    const component = shallow(
      <CheckBox
        iconType="font-awesome"
        iconRight
        uncheckedColor="blue"
        rigth
        center
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
