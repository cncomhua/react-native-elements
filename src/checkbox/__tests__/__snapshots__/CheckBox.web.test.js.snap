// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component should render with icon and checked 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "backgroundColor": "red",
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#bfbfbf"
      name="square-o"
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`CheckBox Component should render with icon and iconRight 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  rigth={true}
  style={
    Object {
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="blue"
      name="square-o"
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`CheckBox Component should render without issues 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#bfbfbf"
      name="square-o"
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: center, title, checked, & onIconPress ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: center, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="#2089dc"
      name="check-square-o"
      onPress={[Function]}
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
    <TextElement
      style={
        Object {
          "color": "#43484d",
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: checkedColor ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: checkedColor ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#22d"
      name="check-square-o"
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: checkedIcon ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: checkedIcon ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#2089dc"
      name="bell-o"
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: checkedTitle, title, checked, & onIconPress ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: checkedTitle, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#2089dc"
      name="check-square-o"
      onPress={[Function]}
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
    <TextElement
      style={
        Object {
          "color": "#43484d",
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      title - is checked
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: component, title, checked, & onIconPress ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: component, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableWithoutFeedback
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#2089dc"
      name="check-square-o"
      onPress={[Function]}
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
    <TextElement
      style={
        Object {
          "color": "#43484d",
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      Title
    </TextElement>
  </View>
</TouchableWithoutFeedback>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: containerStyle ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: containerStyle ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "backgroundColor": "#071",
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#bfbfbf"
      name="square-o"
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: fontFamily ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: fontFamily ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#bfbfbf"
      name="square-o"
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
    <TextElement
      style={
        Object {
          "color": "#43484d",
          "fontFamily": "Courier New",
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: iconRight, title, checked, & onIconPress ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: iconRight, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TextElement
      style={
        Object {
          "color": "#43484d",
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      Title
    </TextElement>
    <Icon
      color="#2089dc"
      name="check-square-o"
      onPress={[Function]}
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: noProps ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: noProps ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#bfbfbf"
      name="square-o"
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: onLongIconPress, title, checked, & onIconPress ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: onLongIconPress, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onLongPress={[Function]}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#2089dc"
      name="check-square-o"
      onPress={[Function]}
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
    <TextElement
      style={
        Object {
          "color": "#43484d",
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: onLongPress, title, checked, & onIconPress ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: onLongPress, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onLongPress={[Function]}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#2089dc"
      name="check-square-o"
      onPress={[Function]}
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
    <TextElement
      style={
        Object {
          "color": "#43484d",
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: onPress, title, checked, & onIconPress ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: onPress, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#2089dc"
      name="check-square-o"
      onPress={[Function]}
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
    <TextElement
      style={
        Object {
          "color": "#43484d",
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: right, title, checked, & onIconPress ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: right, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <Icon
      color="#2089dc"
      name="check-square-o"
      onPress={[Function]}
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
    <TextElement
      style={
        Object {
          "color": "#43484d",
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: textStyle & title ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: textStyle & title ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#bfbfbf"
      name="square-o"
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
    <TextElement
      style={
        Object {
          "backgroundColor": "#071",
          "color": "#43484d",
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: uncheckedColor ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: uncheckedColor ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#d22"
      name="square-o"
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: Props Input: CheckBox: [ prop: uncheckedIcon ] enzyme-shallow: snapshot: Input: CheckBox: [ prop: uncheckedIcon ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Object {
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#bfbfbf"
      name="bell-slash-o"
      size={24}
      style={
        Object {
          "minWidth": 24,
        }
      }
    />
  </View>
</TouchableOpacity>
`;
